torch/csrc/utils/python_arg_parser.h

```diff
@@ -530,4 +556,112 @@ inline PyObject* PythonArgs::pyobject(int i) {
   return args[i];
 }
 
+/*
+ * Stripped down version of PyObject_GetAttrString,
+ * avoids lookups for None, tuple, and List objects,
+ * and doesn't create a PyErr since this code ignores it.
+ *
+ * This can be much faster then PyObject_GetAttrString where
+ * exceptions are not used by caller.
+ *
+ * 'obj' is the object to search for attribute.
+ *
+ * 'name' is the attribute to search for.
+ *
+ * Returns attribute value on success, NULL on failure.
```

Comment:
This is modeled based off some CPython code right? Can we put a cross reference here?

Reply:
It was copied from numpy. [link](https://github.com/numpy/numpy/blob/f4c497c768e0646df740b647782df463825bfd27/numpy/core/src/common/get_attr_string.h#L42)

cc @rgommers

Reply:
Cross reference to numpy then :>